src/mjolnir/graphenhancer.cc

```diff
@@ -652,8 +668,9 @@ bool IsNotThruEdge(GraphReader& reader,
       }
 
       // Add to the end node to expand set if not already visited set
-      if (visitedset.find(diredge->endnode()) == visitedset.end()) {
-        expandset.insert(diredge->endnode());
+      auto endnode = diredge->endnode();
+      if (visitedset.find(endnode) == visitedset.end()) {
```

Comment:
my comment about `.find` was referencing this but my eyes got tricked. i saw visitedset and thought it was expandset..